Python/Include/Sample.hpp

```diff
@@ -60,7 +60,7 @@ namespace Details {
       .def(pybind11::init<typename Sample<R, P...>::Result,
          typename Sample<R, P...>::Parameters>())
       .def(pybind11::init(
-         [](PythonSample s) {
+         [](const PythonSample& s) {
            return Details::sample_cast<Sample<R, P...>>(s);
```

Comment:
This provides the PythonSample -> Sample<R, P...> conversion for each R, P..., it works fine. To do the opposite, for any R, P Sample<R, P...> should be convertible to PythonSample. So I am doing Sample traits + type caster now but honestly, it does not feel like pybind was designed to be used like that to begin with.